### PR TITLE
Provide documentation on configuration profiles/components/add ons.

### PR DESCRIPTION
**Pull request by ericvn**
This PR is for the first paragraph noted in https://github.com/istio/istio.io/issues/3460.

### CODE REVIEW

File: content/docs/setup/kubernetes/additional-setup/config-profile/index.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+---
+title: Configuration Profiles
```
Comment (frankbu) [inline content/docs/setup/kubernetes/additional-setup/config-profile/index.md]: Installation Configuration Profiles

File: content/docs/setup/kubernetes/additional-setup/config-profile/index.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+description: Describe the provided configuration profiles.
+weight: 35
+keywords: [profiles]
+---
+
+This page describes the provided helm configuration profiles. The profiles provide customization of the Istio control plane and of the sidecars for the Istio data plane. You can start with one of Istio’s built-in configuration profiles and then further customize the configuration for your specific needs. The following built-in configuration profiles are currently available:
```
Comment (frankbu) [inline content/docs/setup/kubernetes/additional-setup/config-profile/index.md]:
```suggestion
This page describes the built-in configuration profiles that can be used when
[installing Istio using helm](/docs/setup/kubernetes/install/kubernetes/).
The profiles provide customization of the Istio control plane and of the sidecars for the Istio data plane.
You can start with one of Istio’s built-in configuration profiles and then further customize the configuration for
your specific needs. The following built-in configuration profiles are currently available:
```

File: content/docs/setup/kubernetes/additional-setup/config-profile/index.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`istio-tracing` |   | X |   |   |   |
+| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`grafana` |   | X |   |   |   |
+| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`kiali` |   | X |   |   |   |
+|   |   |   |   |   |   |
+| The authentication version, adding `-auth` adds |   |   |   |   |   |
+| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Control Plane Security |   | X |   |   |   |
+| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;mTLS |   | X |   |   | X |
+| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;SDS |   |   |   |   | X |
+
+To further customize Istio and install addons, you can add one or more `--set <key>=<value>` options in the`helm template` or `helm install` command in the instructions, below. The [Installation Options](/docs/reference/config/installation-options/) lists the complete set of supported installation key and value pairs.
```
Comment (frankbu) [inline content/docs/setup/kubernetes/additional-setup/config-profile/index.md]:
```suggestion
To further customize Istio and install addons, you can add one or more `--set <key>=<value>` options in the`helm template` or `helm install` command that you use when installing Istio. The [Installation Options](/docs/reference/config/installation-options/) lists the complete set of supported installation key and value pairs.
```

File: content/docs/setup/kubernetes/additional-setup/config-profile/index.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+1. **sds**:  used to enable [SDS (secret discovery service) for Istio](/docs/tasks/security/auth-sds). This profile comes only with authentication enabled.
+
+The components marked as **X** are installed within each profile:
+
+| Component | default | demo | minimal | remote | sds |
+| --- | :---: | :---: | :---: | :---: | :---: |
+|Profile filename | `values.yaml` | `values-istio-demo.yaml` | `values-istio-minimal.yaml` | `values-istio-remote.yaml` | `values-istio-sds-auth.yaml` |
+| `istio-citadel` | X | X |   | X | X |
+| Gateway |   |   |   |   |  |
+| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`istio-egressgateway` |   | X |   |   | |
```
Comment (ericvn) [inline content/docs/setup/kubernetes/additional-setup/config-profile/index.md]: Remove galley and istio-gateway from remote profile.

File: content/docs/setup/kubernetes/additional-setup/config-profile/index.md (hunk 1 above)
Comment (ericvn) [reply content/docs/setup/kubernetes/additional-setup/config-profile/index.md]: Updated

File: content/docs/setup/kubernetes/additional-setup/config-profile/index.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
+---
+title: Configuration Profiles
+description: Describe the provided configuration profiles.
```
[the remark this answers is not in the document]
Comment (ericvn) [reply content/docs/setup/kubernetes/additional-setup/config-profile/index.md]: Done

File: content/docs/setup/kubernetes/additional-setup/config-profile/index.md (hunk 3 above)
Comment (ericvn) [reply content/docs/setup/kubernetes/additional-setup/config-profile/index.md]: Done

File: content/docs/setup/kubernetes/additional-setup/config-profile/index.md (hunk 4 above)
Comment (ericvn) [reply content/docs/setup/kubernetes/additional-setup/config-profile/index.md]: Done

File: content/docs/setup/kubernetes/additional-setup/config-profile/index.md (hunk 2 above)
Comment (ericvn) [reply content/docs/setup/kubernetes/additional-setup/config-profile/index.md]: Done

**Comment by ericvn** [comment]
@frankbu Also will take input on table format. I know there was discussion in the issue.

I know _Mixer_ and _Gateway_ are sort of weird as they aren't the _istio-_ named components. This is because the components actually have sub-pods unlike the others.  
 - Maybe we just remove the headers and leave the pods (and resort alphabetically)? 
 - or maybe we add the chart title, like `ingress-gateway` (gateway), `istio-policy` (mixer), etc.?

I tried to separate the `-auth-` and what is there seems OK to me. We could remove the `auth` part of the table and and a Note section:
`values-istio-demo-auth.yaml` enables control plane security and mTLS in addition to the components shown above.
`values-istio-sds-auth.yaml` enables mTLS and SDS.

File: content/docs/setup/kubernetes/additional-setup/config-profile/index.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
+---
+title: Installation Configuration Profiles
+description: Describes the built-in Istio installation configuration profiles.
+weight: 35
+keywords: [profiles]
+---
+
+This page describes the built-in configuration profiles that can be used when
+[installing Istio using helm](/docs/setup/kubernetes/install/kubernetes/).
```
Comment (frankbu) [inline content/docs/setup/kubernetes/additional-setup/config-profile/index.md]:
```suggestion
[installing Istio using helm](/docs/setup/kubernetes/install/helm/).
```

File: content/docs/setup/kubernetes/additional-setup/config-profile/index.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`istio-egressgateway` |   | X |   |   | |
+| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`istio-ingressgateway` | X | X |   |  | X |
+| `istio-galley` | X | X |   |  | X |
+| `istio-pilot` | X | X | X |   | X |
+| Mixer |   |   |   |   |  |
+| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`istio-policy` | X | X |   |   | X |
+| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`istio-telemetry` | X | X |   |   | X |
+| `istio-sidecar-injector` | X | X |   | X | X |
+| `istio-nodeagent` |   |   |   |   | X |
+| Add on |   |   |   |   |   |
```
Comment (frankbu) [inline content/docs/setup/kubernetes/additional-setup/config-profile/index.md]:
```suggestion
| Add ons |   |   |   |   |   |
```

**Comment by frankbu** [comment]
How about remove the `Gateway` and `Mixer` rows, but add one new row at the very top `Core components`, and put everything before the `Add ons` under it.

I'm not crazy about the auth stuff in the table, but since it's at the bottom, I can live it it.

I suggest you make the above tweaks and then we merge it as is. We can get feedback and potentially improve it later, but having it in there would be good so we can refer to it and align all the installation docs around it.

File: content/docs/setup/kubernetes/additional-setup/config-profile/index.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`prometheus` | X | X |   |   | X |
+| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`grafana` |   | X |   |   |   |
+| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`istio-tracing` |   | X |   |   |   |
+| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`kiali` |   | X |   |   |   |
+|   |   |   |   |   |   |
+| The authentication version, adding `-auth` adds |   |   |   |   |   |
+| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Control Plane Security |   | X |   |   |   |
+| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;mTLS |   | X |   |   | X |
```
Comment (frankbu) [inline content/docs/setup/kubernetes/additional-setup/config-profile/index.md]:
```suggestion
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Strict mTLS |   | X |   |   | X |
```

**Comment by rcaballeromx** [comment]
> How about remove the `Gateway` and `Mixer` rows, but add one new row at the very top `Core components`, and put everything before the `Add ons` under it.

I agree, having all the Istio components upfront would be better, each component's name is descriptive enough for readers to know what it is.

> 
> I'm not crazy about the auth stuff in the table, but since it's at the bottom, I can live it it.

Having it at the bottom works for me too.

> 
> I suggest you make the above tweaks and then we merge it as is. We can get feedback and potentially improve it later, but having it in there would be good so we can refer to it and align all the installation docs around it.

Agreed, we can iterate on the table post release.

File: content/docs/setup/kubernetes/additional-setup/config-profile/index.md (hunk 7 above)
Comment (ericvn) [reply content/docs/setup/kubernetes/additional-setup/config-profile/index.md]: Done. Had to add to .spelling and fixed up up another instance on this page.

File: content/docs/setup/kubernetes/additional-setup/config-profile/index.md (hunk 6 above)
Comment (ericvn) [reply content/docs/setup/kubernetes/additional-setup/config-profile/index.md]: Done

File: content/docs/setup/kubernetes/additional-setup/config-profile/index.md (hunk 8 above)
Comment (ericvn) [reply content/docs/setup/kubernetes/additional-setup/config-profile/index.md]: Done

**Comment by ericvn** [comment]
Also made table comments mentioned. I did sort the core components and add-ons in their own sections.

File: .spelling [hunk 9]
```diff
@@ -46,6 +46,10 @@ ack-istio
 ACLs
 Acmeair
 ACS-Engine
+Add-ons
+Add-on
+add-on
```
Comment (frankbu) [inline .spelling]: delete

File: .spelling [hunk 10]
```diff
@@ -46,6 +46,10 @@ ack-istio
...
+add-ons
```
Comment (frankbu) [inline .spelling]: delete

File: .spelling [hunk 11]
```diff
@@ -46,6 +46,10 @@ ack-istio
 ACLs
 Acmeair
 ACS-Engine
+Add-ons
```
Comment (geeknoid) [inline .spelling]: We're been using the spelling 'addon' already (as seen in .spelling). I think we should be consistent one way or another.

File: .spelling [hunk 12]
```diff
@@ -404,6 +408,7 @@ sa
 sayin
 schemas
 SDKs
+sds
```
Comment (geeknoid) [inline .spelling]: Should this be SDS?

File: content/docs/setup/kubernetes/additional-setup/config-profile/index.md [hunk 13]
```diff
@@ -0,0 +1,53 @@
+---
+title: Installation Configuration Profiles
+description: Describes the built-in Istio installation configuration profiles.
+weight: 35
+keywords: [profiles]
```
Comment (geeknoid) [inline content/docs/setup/kubernetes/additional-setup/config-profile/index.md]: Maybe a few more keywords to improve "see also". Look at other files for any keywords that . would make sense here.

Comment (frankbu) [reply content/docs/setup/kubernetes/additional-setup/config-profile/index.md]: install,helm

File: .spelling (hunk 12 above)
Comment (frankbu) [reply .spelling]: I think the profile name should be sds, lowercase, so I think we need this

File: .spelling (hunk 11 above)
Comment (frankbu) [reply .spelling]: addon works for me

Comment (ericvn) [reply .spelling]: I'm fine reverting to addon. Note that the istio.io/content/about/contribute/style-guide/index.md says to use add-on and not addon. We do seem to be inconsistent in the docs.

Comment (ericvn) [reply .spelling]: Per Slack, @geeknoid Will fix the style guide to use addon.

File: .spelling (hunk 12 above)
Comment (ericvn) [reply .spelling]: @frankbu is correct. The lowercase names were what I found and grabbed for the profiles, which meant for consistency I went with `sds`. And thus added it to _.spelling_.

File: content/docs/setup/kubernetes/additional-setup/config-profile/index.md (hunk 13 above)
Comment (ericvn) [reply content/docs/setup/kubernetes/additional-setup/config-profile/index.md]: Done

File: .spelling (hunk 11 above)
Comment (ericvn) [reply .spelling]: Changed to addon

File: .spelling (hunk 10 above)
Comment (ericvn) [reply .spelling]: Done

File: .spelling (hunk 9 above)
Comment (ericvn) [reply .spelling]: Done

File: content/docs/setup/kubernetes/additional-setup/config-profile/index.md [hunk 14]
```diff
@@ -0,0 +1,53 @@
+---
```
Comment (geeknoid) [inline content/docs/setup/kubernetes/additional-setup/config-profile/index.md]: Should the directory be config-profiles?

Comment (ericvn) [reply content/docs/setup/kubernetes/additional-setup/config-profile/index.md]: Done

**Pull request merged**